### ISSUE TITLE
HLine  and  VLine  not updated in a  DynamicMap

### ISSUE THREAD

**Issue by maajdl** [ioam/holoviews#1904]
Hello,

In the snippet below, I try to manipulate the  alpha of  some  HLine  and VLine   in  a DynamicMap.
Unfortunately, on the screen, I can't get anything else than Lines with an  alpha = 1.

Should I try it in another way?

Thanks,

Michel

```
hStyle       = {'HLine':{'style': {'line_width':0, 'color':'black'}}}
vStyle       = {'VLine':{'style': {'line_width':0, 'color':'black'}}}

def test(gr):
    cu = hv.Curve([(0,0),(3,3)])
    hl = hv.Overlay([hv.HLine(x) for x in [0,1,2,3]]+[hv.VLine(x) for x in [0,1,2,3]])
    hgrStyle   = {'HLine':{'style': {'alpha':gr}}}
    vgrStyle   = {'VLine':{'style': {'alpha':gr}}}
    return cu * hl(hgrStyle)(vgrStyle)

dmap = hv.DynamicMap(test, kdims=['alpha'])
slider = [0.0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0]
g = dmap.redim.values(alpha=slider)
g(hStyle)(vStyle)
```
![image](https://user-images.githubusercontent.com/7543475/30689838-7de2fb22-9ec2-11e7-8c5c-de6656c0bad1.png)

**Comment by jlstevens** [comment]
Thanks for reporting this.

Just to be clear, your screenshot shows ``alpha=0`` where you wouldn't expect to see anything anyway. The issue you are experiencing is that you can't see the ``HLine``s and ``VLine``s when moving the alpha slider up to say, ``0.9``?

**Comment by maajdl** [comment]
I see no effect at all of the slider on the chart.
Also, the two alpha values which are indicated do not match except the first time that the sider is moved.
Michel

**Comment by jlstevens** [comment]
I've simplified and improved your example as follows:

```python
line_style = {'line_width':0, 'color':'black'}

def test(gr):
    cu = hv.Curve([(0,0),(3,3)])
    hl = hv.Overlay([hv.HLine(x) for x in [0,1,2,3]]+[hv.VLine(x) for x in [0,1,2,3]])
    alpha_style = dict(line_style, alpha=gr)
    return cu * hl.opts({'HLine':{'style':alpha_style}, 'VLine':{'style':alpha_style}})
```

And I think I can reproduce the issue:

![image](https://user-images.githubusercontent.com/890576/31171096-795d60fe-a8f6-11e7-92c4-3ba6bcd33f5f.png)

But using the ``DynamicMap``:

```python
dmap = hv.DynamicMap(test, kdims=['alpha'])
slider = [0.0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0]
dmap.redim.values(alpha=slider)
```

![image](https://user-images.githubusercontent.com/890576/31171195-ba3017e8-a8f6-11e7-8d30-107a8dbd4527.png)

@philippjfr I would have expected this to work, at least for Bokeh. Is this a bug or a known limitation?

**Comment by philippjfr** [comment]
Fixed by #2013.
